sdk/core/System.ClientModel/src/Options/LoggingOptions.cs

```diff
@@ -0,0 +1,177 @@
+    private string? _clientAssembly;
+    private string? _requestIdHeaderName;
+
+    /// <summary>
+    /// Get or sets value indicating whether HTTP pipeline logging is enabled.
+    /// </summary>
+    public bool IsLoggingEnabled
+    {
+        get => _isLoggingEnabled;
+        set
+        {
+            AssertNotFrozen();
+
+            _isLoggingEnabled = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if request or response content should be logged.
+    /// </summary>
+    public bool IsLoggingContentEnabled
+    {
+        get => _isLoggingContentEnabled;
+        set
+        {
+            AssertNotFrozen();
+
+            _isLoggingContentEnabled = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating maximum size of content to log in bytes. Defaults to 4096.
+    /// </summary>
+    public int LoggedContentSizeLimit
+    {
+        get => _loggedContentSizeLimit;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggedContentSizeLimit = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a list of header names that are not redacted during logging.
+    /// </summary>
+    public IList<string> LoggedHeaderNames
+    {
+        get => _loggedHeaderNames;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggedHeaderNames = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a list of query parameter names that are not redacted during logging.
+    /// </summary>
+    public IList<string> LoggedQueryParameters
+    {
+        get => _loggedQueryParameters;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggedQueryParameters = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the name of the client assembly to include in logging.
+    /// </summary>
+    public string? LoggedClientAssemblyName
+    {
+        get => _clientAssembly;
+        set
+        {
+            AssertNotFrozen();
+
+            _clientAssembly = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the header name that contains the request Id to include in logging.
+    /// </summary>
+    public string? RequestIdHeaderName
+    {
+        get => _requestIdHeaderName;
+        set
+        {
+            AssertNotFrozen();
+
+            _requestIdHeaderName = value;
+        }
+    }
+
+    /// <summary>
+    /// Freeze this instance of <see cref="LoggingOptions"/>.  After this method
+    /// has been called, any attempt to set properties on the instance or call
+    /// methods that would change its state will throw <see cref="InvalidOperationException"/>.
+    /// </summary>
+    public virtual void Freeze()
+    {
+        _frozen = true;
+        _loggedHeaderNames = new ReadOnlyCollection<string>(_loggedHeaderNames.ToArray());
+        _loggedQueryParameters = new ReadOnlyCollection<string>(_loggedQueryParameters.ToArray());
```

Comment:
I believe if we back the public property with a `List<T>` field instead of an `IList<T>` field, there is an `AsReadOnly` method on `List<T>` that you can use instead of creating an instance of the type here.

Reply:
Ah, no, that doesn't work if you want to reassign it to a readonly collection, does it.